### DISCUSSION TITLE
Add an option for Link and Router to prefetch data with getServerSideProps

### DISCUSSION THREAD

**Discussion by arso1er** [vercel/next.js#13105] [Ideas]
Hi,
I was wondering if you could add an option to prefetch data even when a page uses getServerSideProps. Something like
`<Link serverPrefetch={true} href="/post/[id]" as="/post/1"></Link>`

Thanks.

**Comment by w3bdesign** [comment]
Data is not prefetched if you use getServerSideProps, if you use getStaticProps you'll also get data prefetching automatically.

**Reply by arso1er** [reply]
I know that. As it is now getStaticProps is not the most suitable for my use case, so I'd like to be able to prefetch with getServerSideProps.

**Comment by GMaiolo** [comment]
Would be interesting to have prefetch with getServerSideProps at least on link hover or something like that

**Comment by balazsorban44** [comment]
@timneutkens says in https://github.com/vercel/next.js/discussions/11578#discussioncomment-2997 that it would increase the server load. (I assume he means that the default behavior of the Link is to prefetch all links in the viewport). It's true, but getting to prefetch on hover could be a good middle ground, where the user has a high chance of clicking the link, which would mean a few milliseconds head start.

It may also be possible to cancel the prefetch action, if/when the user unhovers the link, but that may be unnecessary.
